### PR TITLE
adds MaxNodes and Start to FindStorageNodes

### PR DESCRIPTION
**Pull request by jenlij**

### CODE REVIEW

File: pkg/overlay/server.go [hunk 1]
```diff
@@ -57,7 +57,11 @@ func (o *Server) BulkLookup(ctx context.Context, reqs *pb.LookupRequests) (*pb.L
 // FindStorageNodes searches the overlay network for nodes that meet the provided requirements
 func (o *Server) FindStorageNodes(ctx context.Context, req *pb.FindStorageNodesRequest) (resp *pb.FindStorageNodesResponse, err error) {
 	opts := req.GetOpts()
-	maxNodes := opts.GetAmount()
+	maxNodes := req.MaxNodes
```
Comment (coyle) [inline pkg/overlay/server.go]: nit -- `req.GetMaxNodes()`

File: pkg/overlay/server.go [hunk 2]
```diff
@@ -67,7 +71,7 @@ func (o *Server) FindStorageNodes(ctx context.Context, req *pb.FindStorageNodesR
 	result := []*pb.Node{}
 	for {
 		var nodes []*pb.Node
-		nodes, start, err = o.populate(ctx, start, maxNodes, restrictedBandwidth, restrictedSpace, excluded)
+		nodes, start, err = o.populate(ctx, req.Start, maxNodes, restrictedBandwidth, restrictedSpace, excluded)
```
Comment (coyle) [inline pkg/overlay/server.go]: nit -- `req.GetStart()`

File: pkg/pb/overlay.proto [hunk 3]
```diff
@@ -59,6 +59,8 @@ message FindStorageNodesRequest {
     int64 objectSize = 1;
     google.protobuf.Duration contractLength = 2;
     OverlayOptions opts = 3;
+    bytes start = 4;
+    int64 maxNodes = 5;
```
Comment (coyle) [inline pkg/pb/overlay.proto]: nit -- int32

File: pkg/overlay/server.go (hunk 2 above)
Comment (coyle) [reply pkg/overlay/server.go]: What happens if Start isn't specified ?

Comment (jenlij) [reply pkg/overlay/server.go]: i think it's fine since it was unspecified originally

**Pull request merged**